interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a library for creating user interfaces. You would use it to automate simple processes. Meaning, without react, you would have to code each instance of a data change and a change with the DOM. It would be similar to logging your starting point and end point of a trip vs, recording your starting point, each stopping point, then the end point of a trip. One would be significantly shorter than the other. React shortens alot of the processes that would otherwise need to be hard coded.
 
-  Researched answer:
+  Researched answer: It is used to handle the layer of an app that will be accessed by users. It is also useful, as it utilizes JSX and a VDOM to shorten many processes of coding. 
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: The render method is always required for class components so that updates from the HTML can be updated to the DOM automatically. 
 
-  Researched answer:
+  Researched answer: The render method helps by displaying code and updates to code, live, without refreshing a webpage. 
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a syntax extension of javascript that allows you to code HTML inside of javascript code. HTML can return multiple elements, while JSX can only return one. 
 
-  Researched answer:
+  Researched answer: HTML is a language, while JSX is an extention of the Javascript language.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a software tool that monitors and modifies software that you code, and allows you to share it. 
 
-  Researched answer:
+  Researched answer: The yarn.lock file is created and nodes packages are updated when yarn is run.
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: An anonymous function is an undefined funtion.
 
-  Researched answer:
+  Researched answer: It provides a way to callback to functions without naming each one. 
 
 
 ## Looking Ahead: Terms for Next Week
 
-1. Conditional rendering:
+1. Conditional rendering: Components that render based on certain arguements being true and/or false.
 
-2. Object-oriented programming:
+2. Object-oriented programming: Method of programming that is structured as a library of parameters that can be reused. Essentially, basing the structure of your entire program on the basis that everything is an object, all inside on big object.
 
-3. Ruby:
+3. Ruby: An open source programming language that is used for front and backend development. 
 
-4. Ruby blocks:
+4. Ruby blocks: Code that can be executed later, through anonymous functions. 
 
-5. Ruby hashes:
+5. Ruby hashes: The equivalent of key:value pairs, but in Ruby. 
```

Comment:
While your answers are satisfactory,  I think you could really improve this area.  #2 was great! For the rest, expand on the info, use examples, and make these more conversational. Remember, these are interview questions, so show off your knowledge!